tests/pipeline/test_technical.py

```diff
@@ -403,3 +406,130 @@ def test_tr_basic(self):
 
         tr.compute(today, assets, out, highs, lows, closes)
         assert_equal(out, np.full((3,), 2.))
+
+
+class MovingAverageConvergenceDivergenceTestCase(ZiplineTestCase):
+    def test_MACD_window_length_generation(self):
+        signal_period = random_integers(1, 90)
+        fast_period = random_integers(signal_period+1, signal_period+100)
+        slow_period = random_integers(fast_period+1, fast_period+100)
+        ewma = MovingAverageConvergenceDivergence(
+            fast_period=fast_period,
+            slow_period=slow_period,
+            signal_period=signal_period,
+        )
+        assert_equal(
+            ewma.window_length,
+            slow_period+signal_period-1,
+        )
+
+    def test_moving_average_convergence_divergence(self):
+        fast_period = 3
+        slow_period = 8
+        signal_period = 2
+
+        macd = MovingAverageConvergenceDivergence(
+            fast_period=fast_period,
+            slow_period=slow_period,
+            signal_period=signal_period,
+        )
+
+        today = pd.Timestamp('2016', tz='utc')
+        nassets = macd.window_length
+        assets = pd.Index(np.arange(nassets))
+        days_col = np.arange(start=-.05,
+                             stop=.01*nassets-.05,
+                             step=.01)[:, np.newaxis]
+        close = np.logspace(start=.01, stop=.10, num=nassets) - 1 + days_col
+
+        dtype = [
+            ('macd', 'f8'),
+            ('signal', 'f8'),
+            ('hist', 'f8'),
+        ]
+        out = np.recarray(
+            shape=(nassets,),
+            dtype=dtype,
+            buf=np.empty(shape=(nassets,), dtype=dtype),
+        )
+        macd.compute(
+            today,
+            assets,
+            out,
+            close,
+            fast_period,
+            slow_period,
+            signal_period,
+        )
+
+        expected_macd = np.array([0.01691553] * nassets)
+        expected_signal = np.array([0.01691553] * nassets)
```

Comment:
Good point, I'll use a different (less efficient) method to calculate MACD and check against that.